pkg/daemon/update.go

```diff
@@ -1809,7 +1813,7 @@ func (dn *Daemon) atomicallyWriteSSHKey(keys string) error {
 	return nil
 }
 
-// Update a given PasswdUser's SSHKey
+// Update a given PasswdUser's SSHKey and Password Hash
```

Comment:
Curious, any reason to do both ssh key and password update in same function?  Keeping them separate would be less error prone because handling defer call during failure will not be needed for both.

Reply:
Yes- it was because the cyclomatic complexity was over limit in (*Daemon).update. 
So I decided to combine it with updateSSHKeys because the functions were similar (they both mock nodes and they both access the passwd user struct).

Reply:
I originally kept it separate but the cyclomatic complexity was already at limit when I started working on this.

Reply:
If we keep the mushed together, we at least need to make sure the passwordhash section is behind the mock (because usermod can't run if there's no node to run against) and the name should reflect what it does (e.g. `UpdatePasswdUsers` or `UpdatePasswdAndSshKeys` or something like that)

Yes, the coreos user check would be duplicated if we split them back out, but that could be a `coreUserExists()` function if we were worried? 

Honestly I'd be okay with just setting `//nolint:gocyclo` on `update()` and leaving UpdatePasswd and UpdateSSH as  separate since the whole point of this is clarity? 

And I feel like that would be clearer than a middle ground like: 

```
func UpdatePasswdUsers() { 

	updatePasswd()
	defer updatePasswd()
	
	updateSSHKeys
	defer updateSSHKeys() 
	
}
```
but that's maybe a matter of opinion?

Reply:
I'm ok with both opinions- not sure what the repercussions of setting `//nolint:gocyclo` on `update()` will do. If its ok then I will be happy to revert because it is more explicit as two separate functions

Reply:
Agree with John :)